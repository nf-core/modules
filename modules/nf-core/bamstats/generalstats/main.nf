process BAMSTATS_GENERALSTATS {
    tag "$meta.id"
    label 'process_single'
    conda "bioconda::bamstats=0.3.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bamstats:0.3.5--he881be0_0':
        'biocontainers/bamstats:0.3.5--he881be0_0' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.json"), emit: json
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    // Several ARGS are available.
    //  -a is a helpfu; one where you can add a BED file
    //  -u, --uniq outputs genomic coverage statistics for uniqely mapped reads
    """
    bamstats \\
        -i $bam \\
        $args \\
        -c $task.cpus \\
        -o ${prefix}.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bamstats: \$(echo \$(bamstats --version 2>&1) | sed 's/^.*bamstats == version://; s/Using.*\$//' | sed 's/built.*//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bamstats: \$(echo \$(bamstats --version 2>&1) | sed 's/^.*bamstats == version://; s/Using.*\$//' | sed 's/built.*//' )
    END_VERSIONS
    """
}

