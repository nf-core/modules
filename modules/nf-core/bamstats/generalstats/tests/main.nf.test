nextflow_process {

    name "Test Process BAMSTATS_GENERALSTATS"
    script "../main.nf"
    process "BAMSTATS_GENERALSTATS"

    tag "modules"
    tag "modules_nfcore"
    tag "bamstats"
    tag "bamstats/generalstats"

    test("sarscov2 - bam") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.json.get(0)[1]).exists() },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.json.get(0)[1]).exists() },
                { assert snapshot(process.out.versions).match("versions_stub") }
            )
        }

    }

}
