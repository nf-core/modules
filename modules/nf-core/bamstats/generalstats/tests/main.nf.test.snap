{
    "versions_stub": {
        "content": [
            [
                "versions.yml:md5,6d858a3f76d1852b879f15436882c3cf"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T16:52:19.198122"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,6d858a3f76d1852b879f15436882c3cf"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T16:52:07.413799"
    }
}